Library/Homebrew/os/mac/mach.rb

```diff
@@ -101,9 +99,8 @@ def mach_data
 
         mach_data << { :arch => arch, :type => type }
       end
+
       mach_data
-    rescue
-      []
```

Comment:
I think this `rescue` has to stay. It not only deals with the `"Not a Mach-O binary."` exception, but also with calling `unpack` on `nil`. Just try `Pathname.new(<file>).mach_data` on a file of length zero.